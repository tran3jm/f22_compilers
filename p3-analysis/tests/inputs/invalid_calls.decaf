int foo;

def bool bar() {
    return true || false;
}

def int main()
{
    bar = true;
	return foo(1, true);
}
