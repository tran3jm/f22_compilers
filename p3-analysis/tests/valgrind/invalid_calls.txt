==2844176== Memcheck, a memory error detector
==2844176== Copyright (C) 2002-2017, and GNU GPL'd, by Julian Seward et al.
==2844176== Using Valgrind-3.15.0 and LibVEX; rerun with -h for copyright info
==2844176== Command: ../decaf inputs/invalid_calls.decaf
==2844176== 
Function 'bar' accessed as a variable on line 9
Invalid call to non-function 'foo' on line 10
Invalid number of function arguments on line 10
==2844176== 
==2844176== HEAP SUMMARY:
==2844176==     in use at exit: 0 bytes in 0 blocks
==2844176==   total heap usage: 1,996 allocs, 1,996 frees, 250,908 bytes allocated
==2844176== 
==2844176== All heap blocks were freed -- no leaks are possible
==2844176== 
==2844176== For lists of detected and suppressed errors, rerun with: -s
==2844176== ERROR SUMMARY: 0 errors from 0 contexts (suppressed: 0 from 0)
