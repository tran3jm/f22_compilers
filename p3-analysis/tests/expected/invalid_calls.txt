Function 'bar' accessed as a variable on line 9
Invalid call to non-function 'foo' on line 10
Invalid number of function arguments on line 10
